### PR TITLE
[cmake/linux] Code coverage

### PR DESCRIPTION
**Pull request by fetzerch**
Now that the tests are working again, I thought it might be a good idea to see what we are testing. 

**Usage:**

```
cmake -DCMAKE_BUILD_TYPE=Coverage <SRC-DIR>/project/cmake`
make coverage
```

Tests are executed as part of the `coverage` or `coverage_xml` target.

**Output:**
unbenannt

I've triggered a build: http://jenkins.kodi.tv/view/Linux/job/LINUX-64/8340 to see if we have the necessary tools installed on Jenkins. ([Cobertura Plugin](https://wiki.jenkins-ci.org/display/JENKINS/Cobertura+Plugin) on the master and `lcov` and `gcovr` on the Linux slave(s)).
Maybe @Memphiz or @wsnipex can help here.

**Known Issue:**
There's one known issue: I had to disable library wrapping in the `Coverage` build type, because there are errors such as:

```
cc: error: __gcov0.-Wl,-wrap,calloc: No such file or directory
```

While this doesn't affect the reports much, it might lead to surprising effects when you're running Kodi built with the `Coverage` type. Just something to be aware of for the moment.
Maybe @notspiff can help to setup `wrapper.def` for this.

**Other platforms:**
This is Linux only for the moment, if anyone is interested to get coverage to work on Windows or OSX, I can assist.

### CODE REVIEW

File: xbmc/network/cddb.cpp [hunk 1]
```diff
@@ -219,8 +219,8 @@ bool Xcddb::queryCDinfo(CCdInfo* pInfo, int inexact_list_select)
 
   //##########################################################
   // Read the data from cddb
-  Recv(false); //erstmal den M�ll abholen
```
Comment (fetzerch) [inline xbmc/network/cddb.cpp]: :laughing: unfortunately gcovr doesn't like non UTF-8 chars.

Comment (Memphiz) [inline xbmc/network/cddb.cpp]: mhh dropping the comment without replacing it with something english? By intention?

File: xbmc/network/cddb.cpp [hunk 2]
```diff
@@ -219,8 +219,8 @@ bool Xcddb::queryCDinfo(CCdInfo* pInfo, int inexact_list_select)
...
-  if ( !Send(read_buffer.c_str()) )
+  Recv(false);
```
Comment (fritsch) [inline xbmc/network/cddb.cpp]: `` `
// Clear pending data on our data connection
```

**Comment by fetzerch** [comment]
Thx, done.

**Comment by wsnipex** [comment]
sadly our jenkins slave on linux is still ubuntu 12.04, which does not provide gcovr...

and I've sent you a small PR that eliminates the need for "make all kodi-test"

**Comment by wsnipex** [comment]
installed gcovr and lcov on the jenkins slave

**Comment by fetzerch** [comment]
Thanks @wsnipex. Squashed your changes with a few minor modifications. (COVERAGE_DEPENDS variable, if logic in main CMakeLists.txt)

Next build triggered: http://jenkins.kodi.tv/view/Helpers/job/LINUX-64/8349/

**Comment by fetzerch** [comment]
@wsnipex: Seems like there's some incompatibility which causes:

```
geninfo: ERROR: /home/jenkins/workspace/LINUX-64/build/build/api1AddonCallbacks_PVR/CMakeFiles/api1AddonCallbacks_PVR.dir/AddonCallbacksPVR.cpp.gcno: reached unexpected end of file
```
(full log: http://jenkins.kodi.tv/view/Linux/job/LINUX-64/8357/console)

I'm not sure if if can be workarounded without upgrading gcc and lcov. I'll still leave the jenkins bits in place, as they don't hurt. But I fear for now it'll only work locally.

**Comment by fetzerch** [comment]
@wsnipex, @hudokkow: Any objections to merging this, even though it doesn't work on jenkins? It might still be useful when working on the tests locally. Currently I don't have time to see if it can be made compatible with 12.04 unfortunately.

**Comment by hudokkow** [comment]
Fine by me.

**Pull request merged**